full-service/src/test_utils.rs

```diff
@@ -361,6 +367,51 @@ pub fn manually_sync_account(
     account
 }
 
+// TODO(cc) refactor so regular and view only account manual sync utilities
+// share some code.
+// Sync view-only-account to most recent block
+pub fn manually_sync_view_only_account(
+    ledger_db: &LedgerDB,
+    wallet_db: &WalletDb,
+    view_only_account_id: &str,
+    logger: &Logger,
+) -> ViewOnlyAccount {
+    let mut account: ViewOnlyAccount;
+    loop {
+        match sync_view_only_account(&ledger_db, &wallet_db, &view_only_account_id, &logger) {
+            Ok(_) => {}
+            Err(SyncError::Database(WalletDbError::Diesel(
+                diesel::result::Error::DatabaseError(kind, info),
+            ))) => {
+                match info.message() {
+                    "database is locked" => log::trace!(logger, "Database locked. Will retry"),
+                    _ => {
+                        log::error!(
+                            logger,
+                            "Unexpected database error {:?} {:?} {:?} {:?} {:?} {:?}",
+                            kind,
+                            info,
+                            info.details(),
+                            info.column_name(),
+                            info.table_name(),
+                            info.hint(),
+                        );
+                        panic!("Could not manually sync account.");
+                    }
+                };
+                std::thread::sleep(Duration::from_millis(500));
+            }
```

Comment:
Optional: Restrict this match further with a guard.
```suggestion
            Err(SyncError::Database(WalletDbError::Diesel(
                diesel::result::Error::DatabaseError(kind, info),
            ))) if info.message() == "database is locked" => {
                log::trace!(logger, "Database locked. Will retry"),
                std::thread::sleep(Duration::from_millis(500));
            },
```